### PR TITLE
Show IPv6 addresses in connection list

### PR DESCRIPTION
**Pull request by glv2**
For issue #8647

### CODE REVIEW

**Comment by jeffro256** [comment]
This PR makes RPC backwards-incompaible for code which assumes "non-empty ip field == IPv4 address". This may or may not be an issue which actually exists, but either way, at least the RPC minor version should be incremented to reflect the change in API.

File: src/daemon/rpc_command_executor.cpp [hunk 1]
```diff
@@ -661,11 +661,18 @@ bool t_rpc_command_executor::print_connections() {
...
     std::string address = info.incoming ? "INC " : "OUT ";
-    address += info.ip + ":" + info.port;
+    if (((epee::net_utils::address_type) info.address_type) == epee::net_utils::ipv6_network_address::get_type_id())
+    {
+      address += "[" + info.ip + "]:" + info.port;
+    }
+    else
+    {
+      address += info.ip + ":" + info.port;
+    }
```
Comment (jeffro256) [inline src/daemon/rpc_command_executor.cpp:671]:
```suggestion
    address += info.address;
```

Also handles Tor/I2P addresses

**Comment by glv2** [comment]
Patch updated.
I hadn't seen this ```info.address``` variable that already has what we want and makes things simpler.
Thanks.

**Review by jeffro256 — APPROVED** [review]
Looks good!

File: src/daemon/rpc_command_executor.cpp [hunk 2]
```diff
@@ -644,7 +644,7 @@ bool t_rpc_command_executor::print_connections() {
     }
   }
 
-  tools::msg_writer() << std::setw(30) << std::left << "Remote Host"
+  tools::msg_writer() << std::setw(74) << std::left << "Remote Host"
```
Comment (selsta) [inline src/daemon/rpc_command_executor.cpp:647]: Would it be possible to keep it at 30 unless an IPv6 peer is connected? It makes the `print_cn` list look weird now and 99% of users won't have any IPv6 connections.

Comment (glv2) [reply src/daemon/rpc_command_executor.cpp:647]: In fact, it's for i2p and onion addresses. They can be much longer than IPv6 addresses.
Is something relying on this field being fixed length? If not, we could make it variable length depending on the current peers' addresses.

Comment (selsta) [reply src/daemon/rpc_command_executor.cpp:647]: >Is something relying on this field being fixed length

Not sure, I did some testing with the following code and it seemed to work well.

```cpp
int width = std::any_of(res.connections.begin(), res.connections.end(), [](const auto &info) {
  return (epee::net_utils::address_type)info.address_type == epee::net_utils::address_type::tor ||
         (epee::net_utils::address_type)info.address_type == epee::net_utils::address_type::i2p ||
         (epee::net_utils::address_type)info.address_type == epee::net_utils::address_type::ipv6;
}) ? 74 : 30;
```

Comment (glv2) [reply src/daemon/rpc_command_executor.cpp]: I updated the patch to adapt the width of the host field to keep it as small as possible.  What do you think?

**Review by selsta — APPROVED** [review]

**Pull request merged**